### ISSUE TITLE
In 6.4: After linking to a dataset, cannot unlink as 'Admin' and error message has code snippets and does not reference objects 

### ISSUE THREAD

**Issue by amberleahey** [IQSS/dataverse#11117]
**What steps does it take to reproduce the issue?**
With the Borealis upgrade to 6.4 in our test environment we found that as a regular 'Admin' after linking a dataset to Dataverse, you cannot unlink and an error appears that has code snippets and does not reference the objects properly. 

*When does this issue occur?
After linking a data, an admin cannot unlink and instead gets an error that has code snippets and does not reference objects and requires translations 

* Which page(s) does it occurs on?
Dataset landing page of the dataset linking/ and unlinking 

* What happens?
An error message appears with code snippets and does not reference the objects and is not human readable and also was not flagged in properties file for translation

* To whom does it occur (all users, curators, superusers)?
all depositors admins, those with permission to link and with 6.4 release I think this was an intended feature to be able to unlink (not just by SuperAdmins)

* What did you expect to happen?

That unlinking would be supported 

**Which version of Dataverse are you using?**

6.4 

**Any related open or closed issues to this bug report?**
some sortof related https://github.com/IQSS/dataverse/issues/10616

**Screenshots:**
![image](https://github.com/user-attachments/assets/11734750-82e1-4379-a275-3943d993dac9)

![image](https://github.com/user-attachments/assets/977ccfb3-4868-4282-b3b2-489f5361b605)

Ideas for improvement as suggested by @JayanthyChengan 
From the IQSS code, it appears that the exception encountered while unlinking (deleteLink) the dataset is appended to the property value “dataset.notlinked.msg”. The exception thrown from the Java code is in English and is not a human-readable error message.

msg = BundleUtil.getStringFromBundle("dataset.notlinked.msg") + ex;

 

The current property value is as follows:
dataset.notlinked.msg=There was a problem linking this dataset to yours:

 

Refer to the code snippet here:
[DatasetPage.java, Line 3633](https://github.com/scholarsportal/dataverse/blob/sp-v6.4/src/main/java/edu/harvard/iq/dataverse/DatasetPage.java#L3633)

 

To improve user experience, we should consider reformatting the “dataset.notlinked.msg” property by removing exception message and adding details about the dataset links, making it more descriptive and human-readable.

 

There might be more cases of this type.

----
**Are you thinking about creating a pull request for this issue?**
Help is always welcome, is this bug something you or your organization plan to fix?

**Comment by pdurbin** [comment]
Hmm, sounds related to this, which shipped in 6.4:

- #10689

@JayanthyChengan thanks for the pointers to the code. @amberleahey thanks for reporting.

**Comment by amberleahey** [comment]
yes thanks, I missed that piece that the error is only for when you are not the owner of the dataset you are linking ! thx! and also, glad to see a fix for matching link permissions coming soon :) 

Here is another screenshot: 
![image](https://github.com/user-attachments/assets/9996a08f-2f4f-44c3-8d32-4116a9d4b1c4)

**Comment by pdurbin** [comment]
Oh, no, sorry, I meant that PR already shipped with 6.4. We'll need a new PR to fix this issue. Do you folks want to make one? 😄

**Comment by amberleahey** [comment]
Not yet! thanks for clarifying!  I'll check in, in the new year!
